bokeh/model.py

```diff
@@ -255,7 +255,21 @@ def _to_json_like(self, include_defaults):
                 that haven't been changed from the default.
 
         """
-        attrs = self.properties_with_values(include_defaults=include_defaults)
+        all_attrs = self.properties_with_values(include_defaults=include_defaults)
+
+        # If __subtype__ is defined, then this model may introduce properties
+        # that don't exist on __view_model__ in bokehjs. Don't serialize such
+        # properties.
+        subtype = getattr(self.__class__, "__subtype__", None)
+        if subtype is not None and subtype != self.__class__.__view_model__:
+            attrs = {}
+            for attr, value in all_attrs.items():
+                if attr in self.__class__.__dict__:
+                    continue
+                else:
+                    attrs[attr] = value
+        else:
+            attrs = all_attrs
```

Comment:
E.g. `Chart` which is `Plot` and is implemented as `Plot` in bokehjs. `Chart` adds `legend`, which previously was serialized. This resulted in an equivalent of `Plot.set("legend", ...)`, which in this PR is not allowed, because `Plot` doesn't define `legend`.